### ISSUE TITLE
Invoke-MsBuild takes a long time to start

### ISSUE THREAD

**Issue by Tdue21** [deadlydog/Invoke-MsBuild#29]
I am currently writing a script for automating my pipeline. I decided to use Invoke-MsBuild for the build part, but discovered that the `Invoke-MsBuild` command seems to hang for a while before starting. 

I do not know if this is a known issue, but I do not think it is the same issue as #11 or #13.

I added some timestamping to my script in order to seem what goes on, and this is the output: 

```
2017-05-14T13:14:55.0336411+02:00 Starting MSBuild.
**********************************************************************
** Visual Studio 2017 Developer Command Prompt v15.0.26430.4
** Copyright (c) 2017 Microsoft Corporation
**********************************************************************
Microsoft (R) Build Engine version 15.1.1012.6693
Copyright (C) Microsoft Corporation. All rights reserved.

Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
Build started 14-05-2017 13:17:15.

...

Time Elapsed 00:00:01.07
2017-05-14T13:17:16.3157579+02:00 MSBuild finished.
```

My code looks like this: 
```
"$(Get-Date -f o) Starting MSBuild."

$result = Invoke-MsBuild -Path $solutionPath -ShowBuildOutputInCurrentWindow -BuildLogDirectoryPath . -KeepBuildLogOnSuccessfulBuilds -LogVerbosity n

"$(Get-Date -f o) MSBuild finished."
```

Apparently something is taking a lot of time, almost 2.5 minute from calling `Invoke-MsBuild` to `MsBuild.exe` actually starting.

**Comment by Tdue21** [comment]
It looks like it may be the VsDevCmd.bat that is taking a lot of time. If that is the case, I suppose there isn't much you can do about it ;)

**Comment by Tdue21** [comment]
I tried creating an empty VsDevCmd.bat in my build folder, and then use the VisualStudioDeveloperCommandPromptFilePath switch to override. This made the build run straight away. 
So, it is the VsDevCmd,bat that is taking a long time to execute. 

I am thinking maybe add a switch to disable calling this?

**Comment by deadlydog** [comment]
Thanks for reporting this and doing the leg-work @Tdue21. I just merged in PR #31 to address this, with a new `-BypassVisualStudioDeveloperCommandPrompt` switch. This feature is in v2.6.0 of the module. If you find any problems with it, let me know. Thanks.

**Comment by Tdue21** [comment]
Seems to work like a charm. +1 for quick response 👍.
